frontend/src/seqvars/components/GenotypeSelect/GenotypeSelect.vue

```diff
@@ -0,0 +1,103 @@
+import { Affected, PedigreeMember, SexAssignedAtBirth } from './types'
+
+const { pedigreeMembers } = defineProps<{ pedigreeMembers: PedigreeMember[] }>()
+
+const membersInheritanceMode = ref(pedigreeMembers.map(() => new Set()))
+
+watch(pedigreeMembers, () => {
+  membersInheritanceMode.value = pedigreeMembers.map(
+    (_, i) => membersInheritanceMode.value.at(i) ?? new Set(),
+  )
+})
+</script>
+
+<template>
+  <CollapsibleGroup title="Genotype">
+    <div style="width: 100%; display: flex; flex-direction: column; gap: 16px">
+      <div style="width: 100%; display: flex; flex-direction: column">
+        <Item
+          v-for="[key, label] in Object.entries(GENOTYPE_LABELS)"
+          :key="key"
+          >{{ label }}</Item
+        >
+      </div>
+
+      <div
+        class="bg-inactive-ui-element"
+        style="margin: 4px 0; width: 100%; height: 1px"
+      ></div>
+
+      <div
+        v-for="(member, index) in pedigreeMembers"
+        :key="member.name"
+        style="display: flex; flex-direction: row; align-items: start; gap: 4px"
+      >
+        <input :id="member.name" type="checkbox" style="margin-top: 6px" />
+        <div style="display: flex; flex-direction: column">
+          <label
+            :for="member.name"
+            style="
+              margin-bottom: 4px;
+              display: flex;
+              align-items: center;
+              gap: 4px;
+            "
+            ><span>{{ member.name }}</span>
+
+            <span
+              :style="
+                member.affected == Affected.UNDEFINED ? { opacity: 0.5 } : {}
+              "
+            >
+              <i-bi-diamond
+                v-if="
+                  member.sexAssignedAtBirth == SexAssignedAtBirth.UNDEFINED &&
+                  member.affected == Affected.UNAFFECTED
+                "
+              />
+              <i-bi-diamond-fill
+                v-if="
+                  member.sexAssignedAtBirth == SexAssignedAtBirth.UNDEFINED &&
+                  member.affected != Affected.UNAFFECTED
+                "
+              />
+
+              <i-bi-circle
+                v-if="
+                  member.sexAssignedAtBirth == SexAssignedAtBirth.FEMALE &&
+                  member.affected == Affected.UNAFFECTED
+                "
+              />
+              <i-bi-circle-fill
+                v-if="
+                  member.sexAssignedAtBirth == SexAssignedAtBirth.FEMALE &&
+                  member.affected != Affected.UNAFFECTED
+                "
+              />
+
+              <i-bi-square
+                v-if="
+                  member.sexAssignedAtBirth == SexAssignedAtBirth.MALE &&
+                  member.affected == Affected.UNAFFECTED
+                "
+              />
+              <i-bi-square-fill
+                v-if="
+                  member.sexAssignedAtBirth == SexAssignedAtBirth.MALE &&
+                  member.affected != Affected.UNAFFECTED
+                "
+              />
```

Comment:
not super happy with the way I had to solve these combined conditions. In React I would've done sth like

```tsx
const { MALE, FEMALE, UNDEFINED: UNDEFINED_SEX } = SexAssignedAtBirth;
const ICONS_BY_SEX_AND_AFFECTED: Record<SexAssignedAtBirth, [Icon, Icon]> = {
  [MALE]: [IconSquare, IconSquareFilled],
  [FEMALE]: [IconCircle, IconCircleFilled],
  [UNDEFINED_SEX]: [IconDiamond, IconDiamondFilled],
};

function Icon({
  sex,
  affected,
}: {
  sex: SexAssignedAtBirth;
  affected: Affected;
}) {
  return React.createComponent(
    ICONS_BY_SEX_AND_AFFECTED[sex][affected == Affected.UNAFFECTED ? 0 : 1],
    affected == Affected.UNDEFINED ? { style: { color: "gray" } } : {}
  );
}
```

which would've been terse and I would have type-level guarantees that I got the mapping right.

Not sure if/how I could express this in Vue

Reply:
Maybe something around `h()`?

https://vuejs.org/guide/extras/render-function.html#creating-vnodes

Reply:
thank you, that nudged me in the right direction, now I'm pretty happy with the solution:
https://github.com/varfish-org/varfish-server/blob/a5052d883dd1c47a615d75158eef3a49b375bd85/frontend/src/seqvars/components/GenotypeSelect/SexAffectedIcon.ts